openedx/core/djangoapps/call_stack_manager/core.py

```diff
@@ -116,29 +119,50 @@ def donottrack(*classes_not_to_be_tracked):
     Returns:
         wrapped function
     """
-
-    def real_donottrack(function):
+    @wrapt.decorator
+    def real_donottrack(wrapped, instance, args, kwargs):  # pylint: disable=W0613
         """takes function to be decorated and returns wrapped function
 
         Args:
             function - wrapped function i.e. real_donottrack
         """
-        def wrapper(*args, **kwargs):
-            """ wrapper function for decorated function
-            Returns:
-                wrapper function i.e. wrapper
-            """
-            if len(classes_not_to_be_tracked) == 0:
-                global TRACK_FLAG  # pylint: disable=W0603
-                current_flag = TRACK_FLAG
-                TRACK_FLAG = False
-                function(*args, **kwargs)
-                TRACK_FLAG = current_flag
-            else:
-                global HALT_TRACKING  # pylint: disable=W0603
-                current_halt_track = HALT_TRACKING
-                HALT_TRACKING = classes_not_to_be_tracked
-                function(*args, **kwargs)
-                HALT_TRACKING = current_halt_track
-        return wrapper
+        if len(classes_not_to_be_tracked) == 0:
+            global TRACK_FLAG  # pylint: disable=W0603
+            current_flag = TRACK_FLAG
+            TRACK_FLAG = False
+            return_value = wrapped(*args, **kwargs)
+            TRACK_FLAG = current_flag
+            return return_value
+        else:
+            global HALT_TRACKING  # pylint: disable=W0603
+            current_halt_track = HALT_TRACKING
+            HALT_TRACKING = classes_not_to_be_tracked
+            return_value = wrapped(*args, **kwargs)
+            HALT_TRACKING = current_halt_track
+            return return_value
     return real_donottrack
+
+
+@wrapt.decorator()
+def trackit(wrapped, instance, args, kwargs):
+    if instance is None:
+        if inspect.isclass(wrapped):
+            # Decorator was applied to a class.
+            # <<<LATER LAST ONE>>>
+            return wrapped(*args, **kwargs)
+        else:
+            # Decorator was applied to a function or staticmethod.
+            # [[[Done]]]
+            capture_call_stack(wrapped.__module__ + "." + wrapped.__name__) #cale -?
```

Comment:
Maybe your best bet would be to pass the function object (`wrapped`) into `capture_call_stack` (and then you can do the same in `CallStackMixin`, or even make `CallStackMixin` just use `trackit` directly)